src/core/unit.c

```diff
@@ -4570,9 +4569,9 @@ int unit_write_setting(Unit *u, UnitWriteFlags flags, const char *name, const ch
         }
 
         if (u->transient_file) {
-                /* When this is a transient unit file in creation, then let's not create a new drop-in but instead
-                 * write to the transient unit file. */
-                fputs_with_newline(data, u->transient_file);
+                /* When this is a transient unit file in creation, then let's not create a new drop-in,
+                 * but instead write to the transient unit file. */
+                fputs_with_newline(u->transient_file, data);
```

Comment:
hmm, but maybe cast the whole thing to (void) if you start to return proper error codes

Reply:
Hmm, the result of `fputs_with_separator` doesn't get casted away either. I don't think it really matters, as the fact that it's a wrapper around `fputs` already indicates that we generally don't care about the result